lib/services/history/server_history_revision.ts

```diff
@@ -0,0 +1,35 @@
+import { BaseHistory, HistoryContent } from '@Lib/services/history/base_history';
+import { HttpResponse } from '../api/http_service';
+import { CreateSourcedPayloadFromObject, RawPayload } from '@Lib/protocol/payloads/generator';
+import { PayloadSource } from '@Lib/protocol/payloads';
+import { SNProtocolService } from '../protocol_service';
+
+export class ServerHistoryRevision extends BaseHistory {
+
+  constructor(content?: HistoryContent) {
+    super(content);
+  }
+
+  static async FromResponse(protocolService: SNProtocolService, itemUuid: string, response?: HttpResponse) {
```

Comment:
Passing a service to a low level object is not good style. Objects should be self contained and should be fed information from a service. This needs to be extracted one level up.